src/components/views/avatars/RoomAvatar.tsx

```diff
@@ -35,29 +35,23 @@ interface IProps extends Omit<ComponentProps<typeof BaseAvatar>, "name" | "idNam
     onClick?(): void;
 }
 
-export function idNameForRoom(room: Room): string {
-    const dmMapUserId = DMRoomMap.shared().getUserIdForRoomId(room.roomId);
-    // If the room is a DM, we use the other user's ID for the color hash
-    // in order to match the room avatar with their avatar
-    if (dmMapUserId) return dmMapUserId;
-
-    if (room instanceof LocalRoom && room.targets.length === 1) {
-        return room.targets[0].userId;
-    }
-
-    return room.roomId;
-}
-
 const RoomAvatar: React.FC<IProps> = ({ room, viewAvatarOnClick, onClick, oobData, size = "36px", ...otherProps }) => {
     const roomName = room?.name ?? oobData?.name ?? "?";
     const avatarEvent = useRoomState(room, (state) => state.getStateEvents(EventType.RoomAvatar, ""));
+    const dmMember = useDmMember(room);
     const roomIdName = useMemo(() => {
-        if (room) {
-            return idNameForRoom(room);
+        if (dmMember) {
+            // If the room is a DM, we use the other user's ID for the color hash
+            // in order to match the room avatar with their avatar
+            return dmMember.userId;
+        } else if (room instanceof LocalRoom && room.targets.length === 1) {
+            return room.targets[0].userId;
+        } else if (room) {
+            return room.roomId;
         } else {
             return oobData?.roomId;
         }
-    }, [oobData, room]);
+    }, [dmMember, oobData, room]);
```

Comment:
Any chance this could be broken out into its own hook with documentation to explain what it does, just as a clean abstraction